openslides/utils/rest_api.py

```diff
@@ -1,20 +1,70 @@
 import re
+from collections import OrderedDict
 
 from urllib.parse import urlparse
 
+from django.core.cache import cache
 from django.core.urlresolvers import reverse
-from rest_framework import response, routers, serializers, viewsets  # noqa
+from rest_framework import permissions, response, routers, serializers, viewsets  # noqa
+from rest_framework.response import Response
 
 from .exceptions import OpenSlidesError
+from .autoupdate import inform_changed_data
 
 router = routers.DefaultRouter()
 
 
+class ModelViewSet(viewsets.ModelViewSet):
+    """
+    Viewset that unses the cache.
+    """
+    # TODO: cache get_serializer_class() because it is now called twice and there
+    #       could be db-queries in it.
+
+    def retrieve(self, request, *args, **kwargs):
+        # Try to get the data from the cache
+        cache_key = self.model.get_rest_cache_key(
+            self.kwargs['pk'],
+            self.get_serializer_class())
+        data = cache.get(cache_key)
+
+        if data is None:
+            # If the data is not in the cache then get it from the db and save it into the cache
+            instance = self.get_object()
```

Comment:
In this case there would always be the database query which I like to get rid of.

I created a new commit in which the original object is created from the cached data and the permission check is made on it.

The best way would be if we do not use check_object_permissions from django_rest_framework but use our own method that works with the serialized data. That should be easy and the fastest way.